src/index.ts

```diff
@@ -30,69 +30,48 @@ const extractTokenFromHeader = (header: string | null): string | null => {
   return token ?? null;
 };
 
-class AuthProvider {
-  constructor(private url: string) {}
-
-  async checkToken(token: string) {
-    const query = gql`
-      query checkToken($token: String!) {
-        checkToken(token: $token) {
-          isValid
-        }
-      }
-    `;
-
-    const result = await request<{ checkToken: { isValid: boolean } }>(
-      this.url,
-      query,
-      {
-        token,
-      }
-    );
-
-    return result.checkToken.isValid;
-  }
-}
-
+// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
 const authProvider = new AuthProvider(process.env.USER_OFFICE_BACKEND!);
 
 async function bootstrap() {
   const serviceList: ServiceEndpoint[] = [
-    // example
-    // the list or the url could come from config / env
-    // { name: 'service-name', url: 'http://localhost:4001' },
     {
       name: 'user-office',
       url: process.env.USER_OFFICE_BACKEND,
-      authCheck: false,
+      includeAuthJwt: false,
     },
     {
       name: 'user-office-scheduler',
       url: process.env.USER_OFFICE_SCHEDULER_BACKEND,
-      authCheck: true,
+      includeAuthJwt: true,
     },
   ];
 
   const gateway = new ApolloGateway({
     serviceList,
 
     buildService(params) {
-      const { url, name, authCheck } = params as ServiceEndpoint;
+      const { url, name, includeAuthJwt } = params as ServiceEndpoint;
 
-      logger.logInfo(`Registering service: '${name}' (${url})`, { authCheck });
+      logger.logInfo(`Registering service: '${name}' (${url})`, {
+        includeAuthJwt,
+      });
 
       return new RemoteGraphQLDataSource<Partial<AppContext>>({
         url,
         willSendRequest({ request, context }) {
-          if (authCheck && context.isValidToken === false) {
-            throw new AuthenticationError('Bad token');
-          }
-
-          if (context.authToken) {
+          if (includeAuthJwt) {
             request.http?.headers.set(
-              'authorization',
-              `Bearer ${context.authToken}`
+              'x-auth-jwt-payload',
```

Comment:
Is there a reason to use a custom HTTP header? also, a quote from https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers "Custom proprietary headers have historically been used with an X- prefix, but this convention was deprecated in June 2012 because of the inconveniences it caused when nonstandard fields became standard"

Reply:
I see, good to know it's deprecated, although I don't think we have to worry about becoming a standard.

I'm using custom header as I don't intend to make any validation on the server side, I trust the information sent by the 
gateway so all I need is the data, so using `authorization` feels inappropriate here. Although I think I should also encode it in base64.